### PR TITLE
moved generate password logic to save method

### PR DESCRIPTION
**Pull request by radik1999**
moved generate password logic to save method. Now to create user we don't need to call method set_password. Password hash will automatically generate during saving

### CODE REVIEW

File: library/catalog/models.py [hunk 1]
```diff
@@ -42,25 +43,35 @@ class MongoUser(Document):
...
+        get_user_model().objects.create_user(username=self.username,
+                                             email=self.email,
+                                             password=django_password)
         return mongo_user
 
     def update(self, **kwargs):
-        mongo_user = super().update(**kwargs)
-        User = get_user_model()
-        User.objects.filter(username=self.username).update(**kwargs)
+        mongo_kwargs = copy(kwargs)
```
Comment (ivan-dubei) [inline library/catalog/models.py:63]: Why do we need to copy kwargs in this case ?

File: library/catalog/models.py [hunk 2]
```diff
@@ -42,25 +43,35 @@ class MongoUser(Document):
...
+        return check_password_hash(self.password, password)
 
     def save(self):
+        if self.password:
+            self.password, django_password = self.generate_passwords(self.password)
         mongo_user = super().save()
-        User = get_user_model()
-        django_user = User(username=self.username, email=self.email, password=self.django_password)
-        django_user.save()
+        get_user_model().objects.create_user(username=self.username,
```
Comment (ivan-dubei) [inline library/catalog/models.py]: It is a bad approach to use bare function calll.
Can be changed like
user_model = get_user_model()

File: library/catalog/models.py [hunk 3]
```diff
@@ -42,25 +43,35 @@ class MongoUser(Document):
...
+            mongo_kwargs['password'], django_kwargs['password'] =\
+                self.generate_passwords(kwargs['password'])
+        mongo_user = super().update(**mongo_kwargs)
+
+        need_to_update_in_django = ['username', 'password', 'email']
+        for field in list(django_kwargs):
+            if field not in need_to_update_in_django:
+                django_kwargs.pop(field)
+
+        get_user_model().objects.filter(username=self.username).update(**django_kwargs)
```
Comment (ivan-dubei) [inline library/catalog/models.py]: The same here
Can be changed like
user_model = get_user_model()

File: library/catalog/models.py (hunk 1 above)
Comment (radik1999) [reply library/catalog/models.py:63]: We need to use a different set of arguments to update mongo and django models. That whay I copy the kwargs for each models and modify them

**Review by ivan-dubei — APPROVED** [review]

**Review by Palusha — APPROVED** [review]

**Pull request merged**